apps/studio/src/components/sidebar/core/table_list/TableListItem.vue

```diff
@@ -252,11 +252,10 @@ export default {
       this.$root.$emit(AppEvent.openTableProperties, { table: this.table })
     },
     pin() {
-      this.$store.dispatch('pins/add', this.table)
-      this.$store.dispatch('updateTableColumns', this.table)
+      this.trigger(AppEvent.togglePinTableList, this.table, true);
     },
     unpin() {
-      this.$store.dispatch('pins/remove', this.table)
+      this.trigger(AppEvent.togglePinTableList, this.table, false);
```

Comment:
Why do we trigger this vs just dispatching to the store directly?

Reply:
Ideally, it's the same as `toggleHideSchema`. I wanted to have an easier tracking which item is pinned or not. We could use a vue `watch` but I feel like we'd need to compare two arrays and then determine which item is missing from the other. With an event, it's known which item immediately. In anyway, I'm very open if you'd want to have a different approach. Just let me know :)

Reply:
Ok, thats fine!